threatmodels/notation-threatmodel.md

```diff
@@ -0,0 +1,89 @@
+# Notation Threat Model
+
+## Overview
+
+Notation is a comprehensive tool for generating and verifying signatures associated with software artifacts. Its purpose is to validate the integrity and authenticity, accomplishing this through the implementation of fine-grained policies.
+
+The following diagram illustrates the architecture and components.
+
+![Notation system overview](../media/notation-system.svg)
+
+- **User**: The actor could be a signer or a verifier. A signer interacts with Notation CLI to sign artifacts. A verifier interacts with Notation CLI to verify artifacts against signatures.
+- **Artifact Builder**: The actor who is responsible for producing software artifacts using build systems. Notation supports signing and verification of software artifacts including but are not limited to container images, helm charts, SBOM (Software Bill of Materials). These artifacts can be stored either in remote registries or locally on disk using [OCI image layout](https://github.com/opencontainers/image-spec/blob/v1.0.0/image-layout.md).
+- **Execution Environment**: The execution environment includes the host and **File System** where notation CLI will be installed and executed. Notation works with a shared responsibility model which means users/organizations are responsible for securing the notation execution environment. Following are the various directories and files used by the Notation:
+  - `config.json` file is used to store various configurations such as credential store information, etc.
+  - `trustpolicy.json` file is used to store trust policy-related data.
+  - `truststore` directory is used to store various trusted certificates used for verifying the signature.
+  - `localkeys` directory is used to store test keys.
+  - `plugins` directory is used to store various plugins.
+  
+  Notation uses credential stores to securely store the registry credentials.
+- **Notation Plugin**: An external component/binary that can integrate as one of the steps in Notation’s workflow for signature generation or verification, see [plugin spec](https://github.com/notaryproject/notaryproject/blob/v1.0.0-rc.2/specs/plugin-extensibility.md) for details.
+- **Registry**: A registry stores container images as well as related content formats, such as Helm charts, OCI artifacts built to the OCI image format specification. Registries are outside Notation trust boundary.
+- **KMS**: A Key Management System that supports the generation and management of signing material such as certificates (along with private key) management or a signing service. KMS are outside Notation trust boundary.
```

Comment:
The definition of KMS is unclear. Here we focus on the signing service.

If a KMS that can do key management but canot use its key to do signing, that KMS is not useful in our current notation remote signing scenario.